### ISSUE TITLE
md-datepicker: Feature - min/max date

### ISSUE THREAD

**Issue by kschwidder** [angular/material#4158]
This is a great first step for a date picker ! Well done !!

It would be awesome if min-date, max-date and a year selection could be added soon. 

BTW: Is it possible to set a date format which can also be entered via keyboard ?

**Comment by mhchen** [comment]
We're working on the min/max date selection already. Currently blocked on #3896 

Year selection is also planned later on.

**Comment by ThomasBurleson** [comment]
@kschwidder - regard *date format*, please submit a separate issue with more details.
